sandbox/libs/authn/src/main/java/org/opensearch/authn/Permissions.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.authn;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * An extension of the abstract Permission class which uses String-object Permissions.
+ *
+ * Example "opensearch.indexing.index.create"
```

Comment:
Should they always be in 4 parts, want to name each of these parts? Checkin with @shanilpa for his thoughts

Reply:
I was personally leaning towards 3 parts but I believe the example in the comment is a current way we do actions using the Security plugin. I will check with @shanilpa like you suggest.